### PR TITLE
Always push EDS after CDS

### PR DESCRIPTION
**Pull request by keithmattix**
Should fix some of the MTLSOrigination flakes we've been seeing

### CODE REVIEW

File: pilot/pkg/xds/delta.go [hunk 1]
```diff
@@ -318,12 +320,35 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
...
 	}
-	return s.pushDeltaXds(con, con.proxy.GetWatchedResource(req.TypeUrl), request)
+
+	err := s.pushDeltaXds(con, con.proxy.GetWatchedResource(req.TypeUrl), request)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case dwr := <-dependentResourceChan:
```
Comment (howardjohn) [inline pilot/pkg/xds/delta.go:330]: I don't think we need this.

IMO we can just do

```
if req.TypeUrl == cluster && proxy.GetWatchedResource(endpoints) != nil {
 ...
}
```

File: pilot/pkg/xds/delta.go [hunk 2]
```diff
@@ -318,12 +320,35 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
...
+			Full:   true,
+			Push:   con.proxy.LastPushContext,
+			Reason: model.NewReasonStats(model.DependentRequest),
+			Start:  con.proxy.LastPushTime,
+			Delta: model.ResourceDelta{
+				// Record sub/unsub, but drop synthetic wildcard info
+				Subscribed: sets.New(dwr.ResourceNames...),
+			},
+		}
+		deltaLog.Debugf("ADS:%s: FORCE DEPENDENT RESPONSE %s for warming.", dwr.TypeUrl, con.conID)
```
Comment (howardjohn) [inline pilot/pkg/xds/delta.go:341]: Lets make this info, should be pretty rare and good to know when it happens

File: pilot/pkg/xds/delta.go [hunk 3]
```diff
@@ -318,12 +320,35 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
...
+	}
+
+	select {
+	case dwr := <-dependentResourceChan:
+		request := &model.PushRequest{
+			Full:   true,
+			Push:   con.proxy.LastPushContext,
+			Reason: model.NewReasonStats(model.DependentRequest),
+			Start:  con.proxy.LastPushTime,
+			Delta: model.ResourceDelta{
```
Comment (howardjohn) [inline pilot/pkg/xds/delta.go:336]: I think this can be empty?

**Review by howardjohn — COMMENTED** [review]
generally LGTM

File: pilot/pkg/xds/delta.go (hunk 3 above)
Comment (keithmattix) [reply pilot/pkg/xds/delta.go:336]: I wondered if empty was permissible; glad to know it is!

File: pilot/pkg/xds/delta.go (hunk 1 above)
Comment (keithmattix) [reply pilot/pkg/xds/delta.go:330]: Ack - I thought we needed to move the watchedResource; good to know there's a func for it

File: pilot/pkg/xds/delta.go (hunk 2 above)
Comment (keithmattix) [reply pilot/pkg/xds/delta.go:341]: Agreed - will do

File: pilot/pkg/xds/delta.go [hunk 4]
```diff
@@ -318,7 +318,41 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
...
+			Full:   true,
+			Push:   con.proxy.LastPushContext,
+			Reason: model.NewReasonStats(model.DependentRequest),
+			Start:  con.proxy.LastPushTime,
+		}
+		deltaLog.Infof("ADS:%s: FORCE DEPENDENT RESPONSE %s for warming.", v3.EndpointType, con.conID)
+		return s.pushDeltaXds(con, dwr, request)
+	}
+
+	return nil
```
Comment (zirain) [inline pilot/pkg/xds/delta.go:355]: if req.TypeUrl != v3.ClusterType {
 return nil
} else if dwr := con.proxy.GetWatchedResource(v3.EndpointType); req.TypeUrl == v3.ClusterType && dwr != nil {
// logic
}

File: pilot/pkg/xds/delta_test.go [hunk 5]
```diff
@@ -295,8 +296,15 @@ func TestDeltaReconnectRequests(t *testing.T) {
 	})
 	s.EnsureSynced(t)
 
-	ads = s.ConnectDeltaADS()
-	// Send initial request
+	ads = s.ConnectDeltaADS().WithTimeout(30 * time.Second) // TODO: Remove before pushing
```
Comment (howardjohn) [inline pilot/pkg/xds/delta_test.go:299]: clean this

**Review by howardjohn — APPROVED** [review]
lgtm put a hold for 2 comments

**Pull request merged**